### DISCUSSION TITLE
GitVersion v6.x.x how do i now access GitVersionInformation?

### DISCUSSION THREAD

**Discussion by conorvahland** [GitTools/GitVersion#3892] [General]
In GitVersion.MSBuild v5.12.0 running on .net6 i can use this `@GitVersionInformation.InformationalVersion` on a razor page to successfully retrieve the version.
After attempting to try out v6.0.0-beta5 it seems it cannot find `GitVersionInformation` any more. I imagine this is related to https://github.com/GitTools/GitVersion/issues/3095 where its been changed from static string to const string back in v6.0.0-alpha.1

In the Visual Studio ObjectBrowser i can see the internal static class exists.
<img width="241" alt="image" src="https://github.com/GitTools/GitVersion/assets/8130526/a00852cc-489d-4228-9df7-4cecbcf5b93b">
however this seems to be completely missing in v6.x.x

am i trying to access this the incorrect way for the new version and its done differently now, or is this something broken?
thanks

**Comment by conorvahland** [comment]
I have updated to GitVersion.MSBuild v6.0.0-beta.7 but still yet to work out a solution to accessing this since the static -> const change.
from reading this page https://gitversion.net/docs/usage/msbuild#how-does-it-work under the heading Accessing Injected Variables it shows how to access these. 

this is what im attempting from the guide but gitVersionInformationType is always null
var assembly = Assembly.GetExecutingAssembly();
var assemblyName = assembly.GetName().Name;
var gitVersionInformationType = assembly.GetType("GitVersionInformation");
var fields = gitVersionInformationType.GetFields();

However, i cannot see these entries on v6.0.0. Can someone please point me in the right direction?

**Reply by asbjornu** [reply]
I can see that #3113 didn't update the documentation, which is unfortunate. Either way, the documentation published on gitversion.net is still for v5, not v6. In #1376 it is suggested that the constants can be consumed as such:

```cs
[assembly: AssemblyConfiguration(GitVersionInformation.PreReleaseLabel)]
```

Does that work for you?

**Reply by conorvahland** [reply]
thanks for the reply. I think its mostly that i dont understand how to use the [assembly:] .... and that i cant manage to verify if the GitVersionInformation class is even being generated. also the `GitVersionInformation`.PreReleaseLabel part doesnt seem to want to resolve in my code editor, hence the complications. v5.x resolved ok

here is my project.AssemblyInfo.cs as it is generated by msbuild in the debug folder.
```
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Reflection;

[assembly: Microsoft.Extensions.Configuration.UserSecrets.UserSecretsIdAttribute("xxxx")]
[assembly: System.Reflection.AssemblyCompanyAttribute("project")]
[assembly: System.Reflection.AssemblyConfigurationAttribute("Debug")]
[assembly: System.Reflection.AssemblyFileVersionAttribute("1.0.0.0")]
[assembly: System.Reflection.AssemblyInformationalVersionAttribute("1.0.0+437dcd6a528741baac390fb4e18da654144d1a1e")]
[assembly: System.Reflection.AssemblyProductAttribute("project")]
[assembly: System.Reflection.AssemblyTitleAttribute("project")]
[assembly: System.Reflection.AssemblyVersionAttribute("1.0.0.0")]

// Generated by the MSBuild WriteCodeFragment class.

```

from my limited understanding, the GitVersionInformation related entries are not being automatically added. but im unsure how to add them to the build process. GitVersion.MSBuild seems to be running fine otherwise

**Reply by conorvahland** [reply]
sorry, my responses should be a tad simpler.
`Does that work for you?`
no it doesnt work for me because i dont know where to add that `assembly` line within my project (.net6.0)
Can you please advise where i would add such entry?
